packages/core/scripts/liquidity-mining/CalculateRollBalancerLPRewards.js

```diff
@@ -0,0 +1,250 @@
+
+  console.log("⚖️  Finding balancer pool info...");
+  // Get the information on a particular pool. This includes a mapping of all previous token holders (shareholders).
+  const pool1Info = await _fetchBalancerPoolInfo(pool1Address);
+  const pool2Info = await _fetchBalancerPoolInfo(pool2Address);
+
+  // Extract the addresses of all historic shareholders.
+  const pool1ShareHolders = pool1Info.shares.flatMap(a => a.userAddress.id);
+  const pool2ShareHolders = pool2Info.shares.flatMap(a => a.userAddress.id);
+  const shareHolders = [...pool1ShareHolders, ...pool2ShareHolders];
+  console.log("🏖  Number of historic liquidity providers:", shareHolders.length);
+
+  let bPool1 = new web3.eth.Contract(poolAbi.abi, pool1Address);
+  let bPool2 = new web3.eth.Contract(poolAbi.abi, pool1Address);
+
+  const shareHolderPayout = await _calculatePayoutsBetweenBlocks(
+    bPool1,
+    bPool2,
+    shareHolders,
+    fromBlock,
+    toBlock,
+    blocksPerSnapshot,
+    umaPerSnapshot,
+    snapshotsToTake
+  );
+
+  console.log("🎉 Finished calculating payouts!");
+  _saveShareHolderPayout(
+    shareHolderPayout,
+    rollNum,
+    fromBlock,
+    toBlock,
+    tokenName,
+    pool1Address,
+    pool2Address,
+    umaPerWeek,
+    blocksPerSnapshot
+  );
+}
+
+// Calculate the payout to a list of `shareHolders` between `fromBlock` and `toBlock`. Split the block window up into
+// chunks of `blockPerSnapshot` and at each chunk assign `umaPerSnapshot` at a prorata basis.
+async function _calculatePayoutsBetweenBlocks(
+  bPool1,
+  bPool2,
+  shareHolders,
+  fromBlock,
+  toBlock,
+  blockPerSnapshot,
+  umaPerSnapshot
+) {
+  // Create a structure to store the payouts for all historic shareholders.
+  let shareHolderPayout = {};
+  for (shareHolder of shareHolders) {
+    shareHolderPayout[shareHolder] = toBN("0");
+  }
+
+  console.log("🏃‍♂️Iterating over block range and calculating payouts...");
+
+  // create new progress bar to show the status of blocks traversed.
+  const progressBar = new cliProgress.SingleBar(
+    {
+      format: "[{bar}] {percentage}% | snapshots traversed: {value}/{total}"
+    },
+    cliProgress.Presets.shades_classic
+  );
+  progressBar.start(Math.ceil((toBlock - fromBlock) / blockPerSnapshot), 0);
+  for (currentBlock = fromBlock; currentBlock < toBlock; currentBlock += blockPerSnapshot) {
+    shareHolderPayout = await _updatePayoutAtBlock(bPool1, bPool2, currentBlock, shareHolderPayout, umaPerSnapshot);
+    progressBar.update(Math.ceil((currentBlock - fromBlock) / blockPerSnapshot) + 1);
+  }
+  progressBar.stop();
+
+  return shareHolderPayout;
+}
+
+// For a given `blockNumber` (snapshot in time), return an updated `shareHolderPayout` object that has appended
+// payouts for a given `bPool` at a rate of `umaPerSnapshot`.
+async function _updatePayoutAtBlock(bPool1, bPool2, blockNumber, shareHolderPayout, umaPerSnapshot) {
+  // Get the total supply of Balancer Pool tokens at the given snapshot's block number.
```

Comment:
As discussed offline, we probably want to weight by the number of yUSD tokens that these balances represent.

Reply:
Ah this is a great point.